FIX-GENERIC-00002

DennisJensen-OWC opened the pull request:
Move the Profile Specific Release directory data into the appropriate Profile directory.  Add a build-tool that copies this release information from the appropriate Profile Release directory into the common Release directory during build as part of the make checkout: –and– update: just after the point where the data has been updated.

A checkout prior to this point will yield:
* The Common Release directory contains the release information for multiple Profiles making it non-generic
* The Release directory was laid out in Version Product Build order instead of Product Version Build order, which is more intuitive because generally a Build belongs to a Version, which belongs to a Product

jordanhubbard commented:
Just to understand:  The tool just uses the profile release directory as a source for the README/ReleaseNotes/etc and copies that into the "generated build product" so that it doesn't get skipped by mistake (that would be a nice enhancement - we've always done that by hand, and often forget)?  What about "Nightlies" - those simply don't match the specific VERSION and are a no-op for this process?

DennisJensen-OWC commented:
Okay I added to the end of Checkout and to the end of Update within the Makefile.inc1 as those were the only two recipes that seemed to pull changes locally to the project so they would only need to be copied over if they did not already exists there or what existed there was out of date due to changes that had been made.  If neither of these scenarios are part of your “Nightlies” then we would need to add the one liner that I added to the Checkout: and Update: recipes to whatever recipe you use for “Nightlies”


> On Dec 20, 2016, at 1:03 PM, Jordan Hubbard <notifications@github.com> wrote:
> 
> Just to understand: The tool just uses the profile release directory as a source for the README/ReleaseNotes/etc and copies that into the "generated build product" so that it doesn't get skipped by mistake (that would be a nice enhancement - we've always done that by hand, and often forget)? What about "Nightlies" - those simply don't match the specific VERSION and are a no-op for this process?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/freenas/freenas-build/pull/23#issuecomment-268328208>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AV1Pn1rCHc3qyTjIIvuxqzWuOQr3SuiAks5rKCaAgaJpZM4LSCju>.
>

DennisJensen-OWC commented:
So is this going to get merged before a conflict is created ? :)  Or is something else needing to be done with this?
